app/controllers/sessions_controller.rb

```diff
@@ -63,6 +67,8 @@ def failure
   private
 
   def pre_login_destination
-    session[:pre_login_destination]
+    destination = session[:pre_login_destination]
+    session.delete :pre_login_destination
+    return destination
```

Comment:
Need to do a bit of cleanup of the pre_login_destination, otherwise it will always redirect to the accounts page